rotator/rotator.go

```diff
@@ -0,0 +1,223 @@
+package rotator
+
+import (
+	"time"
+
+	awsTools "github.com/mattermost/rotator/aws"
+	k8sTools "github.com/mattermost/rotator/k8s"
+	"github.com/mattermost/rotator/model"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"k8s.io/client-go/kubernetes"
+)
+
+// AutoscalingGroup creates a autoscaling group object.
+type AutoscalingGroup struct {
+	Name            string
+	DesiredCapacity int
+	Nodes           []string
+}
+
+// RotatorMetadata is a container struct for any metadata related to cluster rotator.
+type RotatorMetadata struct {
+	MasterGroups []AutoscalingGroup `json:"MasterGroups,omitempty"`
+	WorkerGroups []AutoscalingGroup `json:"WorkerGroups,omitempty"`
+}
+
+// InitRotateCluster is used to call the RotateCluster function.
+func InitRotateCluster(cluster *model.Cluster, rotatorMetadata *RotatorMetadata, logger *logrus.Entry) (*RotatorMetadata, error) {
```

Comment:
@gabrieljackson We now have a custom logger entry here and provisioner PR updated to pass its own logger in the rotator ;)